### PR TITLE
Dashboard settings form QOL improvements

### PR DESCRIPTION
**Pull request by zahid47**
addresses #49 

`shortBio` is marked as optional in the Prisma schema, so the initial value is `null` which doesn't match with the type `CurrentUser`. So I'm setting it as an empty string if it is null. Not sure if it is the best approach, but any advice is appreciated!

### CODE REVIEW

**Comment by moinulmoin** [comment]
Thanks for the PR. I will check soon

File: src/types/index.ts [hunk 1]
```diff
@@ -29,15 +29,14 @@ export const settingsSchema = z.object({
...
-    })
+    .string()
     .max(150, {
       message: "Short bio must be at most 150 characters.",
     })
     .min(10, {
       message: "Short bio must be at least 10 characters.",
-    }),
+    })
+    .or(z.literal("")),
```
Comment (moinulmoin) [inline src/types/index.ts:39]: you can just call optinal method at the end to make it optional. take a look here

https://zod.dev/?id=optionals

**Review by moinulmoin — CHANGES_REQUESTED** [review]
you can just call optinal method at the end to make it optional. take a l

File: src/types/index.ts (hunk 1 above)
Comment (moinulmoin) [inline src/types/index.ts:39]:
```suggestion
  shortBio: z
    .string()
    .max(150, {
      message: "Short bio must be at most 150 characters.",
    })
    .min(10, {
      message: "Short bio must be at least 10 characters.",
    })
    .optional(),
```

File: src/app/dashboard/settings/settings-form.tsx [hunk 2]
```diff
@@ -48,7 +48,7 @@ export default function SettingsForm({
     values: {
       name: currentUser.name,
       email: currentUser.email,
-      shortBio: currentUser.shortBio,
+      shortBio: currentUser.shortBio || "",
```
Comment (moinulmoin) [inline src/app/dashboard/settings/settings-form.tsx:51]:
```suggestion
      shortBio: currentUser.shortBio,
```

File: src/components/layout/image-upload-modal.tsx [hunk 3]
```diff
@@ -123,7 +124,7 @@ export default function ImageUploadModal({
                   loading="lazy"
                 />
               </div>
-              <div className=" mt-10">
+              <div className="mt-10 flex">
```
Comment (moinulmoin) [inline src/components/layout/image-upload-modal.tsx:127]:
```suggestion
              <div className="mt-10">
```

**Review by moinulmoin — CHANGES_REQUESTED** [review]

File: src/app/dashboard/settings/settings-form.tsx [hunk 4]
```diff
@@ -187,15 +187,18 @@ export default function SettingsForm({
           )}
         />
 
-        <div>
+        <div className="flex">
```
Comment (moinulmoin) [inline src/app/dashboard/settings/settings-form.tsx:190]:
```suggestion
        <div>
```

**Comment by moinulmoin** [comment]
All good. just need a few changes and then ready to merge. Thanks

**Comment by zahid47** [comment]
> you can just call optinal method at the end to make it optional. take a look here
> 
> https://zod.dev/?id=optionals

This was my first intuition as well. But it doesn't work if you type something in the short bio field and then remove it. Because then it becomes an empty string (""), but it won't let you update your profile with bio as an empty string as min length is 10 chars.

And the flex class in the other file is to fix the alignment with "cancel" button. During loading state the alignment gets messed up because of the loading icon

**Comment by moinulmoin** [comment]
> And the flex class in the other file is to fix the alignment with "cancel" button. During loading state the alignment gets messed up and causes layout shift because of the loading icon

Hey, 

> > you can just call optinal method at the end to make it optional. take a look here
> > https://zod.dev/?id=optionals
> 
> This was my first intuition as well. But it doesn't work if you type something in the short bio field and then remove it. Because then it becomes an empty string (""), but it won't let you update your profile with bio as an empty string as min length is 10 chars.
> 
> And the flex class in the other file is to fix the alignment with "cancel" button. During loading state the alignment gets messed up and causes layout shift because of the loading icon

Hey, seems you are right, I removed the min and max and I noticed font flickering issue while checking the buttons layout shift. I added few adjustments. thanks.

**Review by moinulmoin — APPROVED** [review]

**Pull request merged**